### ISSUE TITLE
FAILED: Unexpected message in HttpServerSpec

### ISSUE THREAD

**Issue by ktoso** [akka/akka-http#595]
https://jenkins.akka.io:8498/job/pr-validator-akka-http/371/testReport/junit/akka.http.impl.engine.server/HttpServerSpec/The_server_implementation_should_support_request_timeouts_which_are_programmatically_increased__not_expiring_/

**Comment by ortigali** [comment]
This failure may occur when it is too late to [updateTimeout](https://github.com/akka/akka-http/blob/5932237a86a432d623fafb1e84eeeff56d7485fe/akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerSpec.scala#L845). because of [here](https://github.com/akka/akka-http/blob/5932237a86a432d623fafb1e84eeeff56d7485fe/akka-http-core/src/main/scala/akka/http/impl/engine/server/HttpServerBluePrint.scala#L337) (maybe GC pause influence it)

I could not find how to know updateTimeout is successful or not. 
I propose adding method to TimeoutAccess class to get current timeout
``` scala
  def timeout: Future[Duration] // this is hard to use
  // or
  def timeout: Duration  // this is good, but with locking
```

After updateTimeout we can check timeout changed or not by TimeoutAccess.timeout method.

If this proposal is correct, may I try it?

**Comment by jlprat** [comment]
Another example, if needed: https://jenkins.akka.io:8498/job/pr-validator-akka-http/745/testReport/junit/akka.http.impl.engine.server/HttpServerSpec/The_server_implementation_should_support_request_timeouts_which_are_programmatically_increased__not_expiring_/

**Comment by jrudolph** [comment]
https://jenkins.akka.io:8498/job/pr-validator-akka-http/1221/testReport/junit/akka.http.impl.engine.server/HttpServerSpec/The_server_implementation_should_support_request_timeouts_which_are_programmatically_increased__not_expiring_/

**Comment by jonas** [comment]
https://jenkins.akka.io:8498/job/pr-validator-akka-http/1226/testReport/junit/akka.http.impl.engine.server/HttpServerSpec/The_server_implementation_should_support_request_timeouts_which_are_programmatically_increased__not_expiring_/

**Comment by raboof** [comment]
https://jenkins.akka.io:8498/job/pr-validator-akka-http/1255/

**Comment by raboof** [comment]
@ortigali thanks for taking the time to look into this. Exposing the actual timeout in `TimeoutAccess` might work, but then we're testing based on inspecting internals - I like how we test externally visible behavior here.

Perhaps we should simply choose less tight timings?

It'd be really cool if we could get more control over the way time is handled instead of relying on the system clock, but that might be tough.

**Comment by jrudolph** [comment]
https://jenkins.akka.io:8498/job/pr-validator-akka-http/1983/testReport/junit/akka.http.impl.engine.server/HttpServerSpec/The_server_implementation_should_support_request_timeouts_which_are_programmatically_increased__not_expiring_/

**Comment by jrudolph** [comment]
Fixed by #1221.
